### DISCUSSION TITLE
Few improvements to CollapsingState

### DISCUSSION THREAD

**Discussion by 4JX** [emilk/egui#1566] [Ideas]
I've made two patches to `CollapsingState` to add some new functionality:
- Changing the side the collapser is drawn at
- Custom icon support 

However, I'd like to get a check on that second feature (https://github.com/4JX/egui/commit/73d520a0f6f158bc26050c4940f8344e3fe29961), since it does some possibly breaking changes (`Fn` should still suffice for 100% of icon functions, but just in case) before opening a PR.

I also wanted to implement a "no icon" mode for `CollapsingState`, where it would not ever call the draw icon function. The issue is that I would have to introduce an `Option` inside the `HeaderResponse.toggle_button_response`, making the usage of that response slightly unergonomic. Do you think it'd be a problem to do it this way?

**Comment by emilk** [comment]
Breaking changes are fine (even though we of course try to minimize them). I think this will be easier to discuss if you open a draft PR!
